packages/components/pie-radio/test/component/pie-radio.spec.ts

```diff
@@ -427,106 +389,93 @@ test.describe('PieRadio - Component tests', () => {
+            await page.getByTestId(radio.selectors.input.dataTestId).click();
 
-            await page.evaluate(() => {
-                const radio = document.querySelector('pie-radio') as PieRadio;
+            await page.getByTestId(radio.selectors.container.dataTestId).evaluate((el) => {
+                const radio = el as PieRadio;
                 radio.checked = false;
                 return radio;
             });
 
-            await page.click('button[type="submit"]');
+            await page.locator('pie-button', { hasText: 'Submit' }).click();
 
-            const formDataObj = await getFormDataObject(page, '#formDataJson');
+            const formDataOutput = await page.locator('#formDataOutput').textContent();
 
             // Assert
-            expect(formDataObj).toStrictEqual({});
+            expect(formDataOutput).toStrictEqual('{}');
         });
 
         [true, false].forEach((checked) => {
             test(`should not submit the value if the input is disabled when checked is ${checked}`, async ({ page }) => {
                 // Arrange
-                await page.setContent(`
-                    <form id="testForm" action="/foo" method="POST">
-                        <pie-radio
-                            name="testName"
-                            value="testValue"
-                            disabled
-                            ${checked ? 'checked' : ''}>
-                        </pie-radio>
-                        <button type="submit">Submit</button>
-                    </form>
-                    <div id="formDataJson""></div>
-                `);
-                await setupFormDataExtraction(page, '#testForm', '#formDataJson');
+                const props : RadioProps = {
+                    checked,
+                    disabled: true,
+                    value: 'testValue',
+                    name: 'testName',
+                };
+
+                const radioFormPage = new BasePage(page, 'radio--example-form');
+                await radioFormPage.load({ ...props });
 
                 // Act
-                await page.click('button[type="submit"]');
-                const formDataObj = await getFormDataObject(page, '#formDataJson');
+                await page.locator('pie-button', { hasText: 'Submit' }).click();
+                const formDataOutput = await page.locator('#formDataOutput').textContent();
 
                 // Assert
-                expect(formDataObj).toStrictEqual({});
+                expect(formDataOutput).toStrictEqual('{}');
             });
         });
 
         [true, false].forEach((checked) => {
-            test(`should not submit the value inside a disabled fieldset when checked is ${checked}`, async ({ page }) => {
+            // Can't get this to work for som reason, disabled fieldset doesn't cause the radio to be disabled
+            test.skip(`should not submit the value inside a disabled fieldset when checked is ${checked}`, async ({ page }) => {
```

Comment:
I'm unsure why I can't get this working. Despite the `ExampleFormDisabledFieldsetTemplate` defining a disabled fieldset, the underlying `pie-radio` isn't disabled.

I've also copied the old code into the Storybook template, and that also wasn't disabled, so I'm unsure how it was passing before.

I'm not sure if this logic is correct? - https://github.com/justeattakeaway/pie/blob/main/packages/components/pie-radio/src/index.ts#L66

Any ideas @jamieomaguire?

Reply:
Had something similar in the switch refactors. Have you tried passing in the disabled prop through the load function?

Reply:
The goal of this test is to check that even if the radio isn't in a disabled set, setting `disabled` on a parent `fieldset` element will cause the radio to be disabled.

Reply:
Just caught up with @jamieomaguire and we agreed to delete this test as in the context of a fieldset, you'd opt to use `pie-radio-group` instead